ara/ui/templatetags/truncatepath.py

```diff
@@ -40,9 +40,17 @@ def truncatepath(path, count):
     if len(path) < length:
         return path
 
+    if "/" not in path:
+        # Non-paths look better with the end cut off
+        return path[: length - 4] + "..."
+
     dirname, basename = os.path.split(path)
+
+    if len(basename) >= length:
+        return "..." + basename[4 - length:]
```

Comment:
@dmsimard Well, this is fun.

```
2021-04-15 21:09:21.590455 | fedora-32 | ##################################################
2021-04-15 21:09:21.590466 | fedora-32 | # flake8
2021-04-15 21:09:21.590476 | fedora-32 | ##################################################
2021-04-15 21:09:25.195770 | fedora-32 | /home/zuul/src/github.com/ansible-community/ara/ara/ui/templatetags/truncatepath.py:50:43: E203 whitespace before ':'
2021-04-15 21:09:25.195921 | fedora-32 |         return "..." + basename[4 - length :]
2021-04-15 21:09:25.195939 | fedora-32 |                                           ^
```

```
2021-04-15 21:38:27.783629 | fedora-32 | ##################################################
2021-04-15 21:38:27.783649 | fedora-32 | # black
2021-04-15 21:38:27.783657 | fedora-32 | ##################################################
2021-04-15 21:38:31.585030 | fedora-32 | --- /home/zuul/src/github.com/ansible-community/ara/ara/ui/templatetags/truncatepath.py	2021-04-15 21:33:24.220000 +0000
2021-04-15 21:38:31.585070 | fedora-32 | +++ /home/zuul/src/github.com/ansible-community/ara/ara/ui/templatetags/truncatepath.py	2021-04-15 21:38:31.579235 +0000
2021-04-15 21:38:31.585090 | fedora-32 | @@ -45,11 +45,11 @@
2021-04-15 21:38:31.585098 | fedora-32 |          return path[: length - 4] + "..."
2021-04-15 21:38:31.585120 | fedora-32 |
2021-04-15 21:38:31.585127 | fedora-32 |      dirname, basename = os.path.split(path)
2021-04-15 21:38:31.585138 | fedora-32 |
2021-04-15 21:38:31.585144 | fedora-32 |      if len(basename) >= length:
2021-04-15 21:38:31.585150 | fedora-32 | -        return "..." + basename[4 - length:]
2021-04-15 21:38:31.585155 | fedora-32 | +        return "..." + basename[4 - length :]
2021-04-15 21:38:31.585177 | fedora-32 |
2021-04-15 21:38:31.585183 | fedora-32 |      while dirname:
2021-04-15 21:38:31.585189 | fedora-32 |          if len(dirname) + len(basename) + 4 < length:
2021-04-15 21:38:31.585195 | fedora-32 |              break
2021-04-15 21:38:31.585200 | fedora-32 |          dirlist = dirname.split("/")
2021-04-15 21:38:31.586050 | fedora-32 | would reformat /home/zuul/src/github.com/ansible-community/ara/ara/ui/templatetags/truncatepath.py
```

Reply:
@flowerysong oh no, flake8 and black disagreeing ?

It's totally fine to quiet down black with something like this: https://github.com/ansible-community/ara/blob/e4cd2153c935a56ac1d201a3da74024214173e3f/ara/cli/playbook.py#L182-L188